Library/Homebrew/diagnostic.rb

```diff
@@ -785,6 +785,22 @@ def check_coretap_git_origin
               git -C "#{coretap_path}" remote set-url origin #{Formatter.url("https://github.com/Homebrew/homebrew-core.git")}
           EOS
         end
+
+        head = coretap_path.git_head
+        return unless head && head !~ %r{refs/heads/master}
+
+        <<-EOS.undent
+          Suspicious #{CoreTap.instance} git head found.
+
+          With a non-standard head, your local version of Homebrew might not
+          have all of the changes intended for the most recent release. The
+          current git head is:
+            #{head}
+
+          Unless you have compelling reasons, consider setting the head to
+          point at the master branch by running:
+            git checkout master
```

Comment:
Replace all the above text with something like:

```
Homebrew/homebrew-core is not on the master branch

Check out the master branch by running:
  git -C "$(brew --repo homebrew/core)" checkout master
```

Want to avoid Git jargon where possible here.